cognite_toolkit/_cdf_tk/loaders/_resource_loaders/agent_loaders.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+from collections.abc import Hashable, Iterable, Sequence
+from functools import lru_cache
+from pathlib import Path
+from typing import Any
+
+from cognite.client.data_classes.capabilities import Capability
+from cognite.client.exceptions import CogniteAPIError
+from cognite.client.utils.useful_types import SequenceNotStr
+
+from cognite_toolkit._cdf_tk._parameters.constants import ANY_INT, ANYTHING
+from cognite_toolkit._cdf_tk._parameters.data_classes import ParameterSpec, ParameterSpecSet
+from cognite_toolkit._cdf_tk.client.data_classes.agents import Agent, AgentList, AgentWrite, AgentWriteList
+from cognite_toolkit._cdf_tk.loaders._base_loaders import ResourceLoader
+
+
+class AgentLoader(ResourceLoader[str, AgentWrite, Agent, AgentWriteList, AgentList]):
+    folder_name = "agents"
+    filename_pattern = r".*\.Agent$"  # Matches all yaml files whose stem ends with '.Agent'.
+    resource_cls = Agent
+    resource_write_cls = AgentWrite
+    list_cls = AgentList
+    list_write_cls = AgentWriteList
+    kind = "Agent"
+    _doc_base_url = ""
+    _doc_url = "https://pr-2829.specs.preview.cogniteapp.com/20230101-alpha.json.html"
+
+    @classmethod
+    def get_id(cls, item: AgentWrite | Agent | dict) -> str:
+        if isinstance(item, dict):
+            return item["externalId"]
+        return item.external_id
+
+    @classmethod
+    def dump_id(cls, id: str) -> dict[str, Any]:
+        return {"externalId": id}
+
+    @classmethod
+    def get_required_capability(
+        cls, items: Sequence[AgentWrite] | None, read_only: bool
+    ) -> Capability | list[Capability]:
+        return []
+
+    def load_resource_file(
+        self, filepath: Path, environment_variables: dict[str, str | None] | None = None
+    ) -> list[dict[str, Any]]:
+        loaded = super().load_resource_file(filepath, environment_variables)
+        for item in loaded:
+            # the API always returns instructions
+            if "instructions" not in item:
```

Comment:
Hmm, suggest not doing this and handling in the `.dump_resource` instead. As this will have push an empty instructions into the agent config if you do cdf modules pull.